src/validation.h

```diff
@@ -15,6 +15,7 @@
 #include <chain.h>
 #include <chainparams.h>
 #include <consensus/amount.h>
+#include <consensus/params.h>
```

Comment:
0fc617a406232e21bdc8b39b2f8308f965775825 Any reason to add this include when the existing fwd-decl is enough?

Reply:
Adding `#include <chainparams.h>` in the previous commit already brings `consensus/params.h` in so the fwd declaration isn't saving anything. Later, `DeploymentActive*` use info from `consensus/params.h` too.

Reply:
I think for passing references or pointers, a fwd-decl is enough.

iwyu seems to agree:


```
validation.h should remove these lines:
- #include <config/bitcoin-config.h>  // lines 10-10
- #include <consensus/params.h>  // lines 18-18
- #include <stdint.h>  // lines 39-39
- #include <util/hasher.h>  // lines 30-30
- class CBlockTreeDB;  // lines 46-46
- class CTxMemPool;  // lines 47-47
- struct AssumeutxoData;  // lines 53-53
- struct ChainTxData;  // lines 49-49
- struct DisconnectedBlockTransactions;  // lines 50-50
- struct LockPoints;  // lines 52-52

Reply:
Ah, iwyu treats the `DeploymentActive` stuff as recommending that deploymentstatus.h add the `#include <consensus/params.h>` not validation.h. I've got some later commits that move those functions into validation.h which updates iwyu's recommendations to a #include.

Since it's already included indirectly by chainparams.h I don't really see the include as making it worse, but it's not really better either, so will change it back if there are other fixes to bundle it with.

Reply:
Done now